extensions/azurecore/src/azureResource/services/subscriptionService.ts

```diff
@@ -23,4 +39,33 @@ export class AzureResourceSubscriptionService implements IAzureResourceSubscript
 
 		return subscriptions;
 	}
+
+	/**
+	 * Gets all subscriptions for all tenants of the given account. Any errors that occur while fetching the subscriptions for each tenant
+	 * will be displayed to the user, but this function will only throw an error if it's unable to fetch any subscriptions.
+	 * @param account The account to get the subscriptions for
+	 * @returns The list of all subscriptions on this account that were able to be retrieved
+	 */
+	public async getAllSubscriptions(account: AzureAccount): Promise<azureResource.AzureResourceSubscription[]> {
+		const subscriptions: azureResource.AzureResourceSubscription[] = [];
+		let gotSubscriptions = false;
+		const errors: Error[] = [];
+
+		for (const tenant of account.properties.tenants) {
+			try {
+				const token = await azdata.accounts.getAccountSecurityToken(account, tenant.id, azdata.AzureResource.ResourceManagement);
+				subscriptions.push(...(await this.getSubscriptions(account, new TokenCredentials(token.token, token.tokenType), tenant.id) || <azureResource.AzureResourceSubscription[]>[]));
+				gotSubscriptions = true;
+			} catch (error) {
+				const errorMsg = localize('azure.resource.tenantSubscriptionsError', "Failed to get subscriptions for account {0} (tenant '{1}'). {2}", account.key.accountId, tenant.id, AzureResourceErrorMessageUtil.getErrorMessage(error));
```

Comment:
should we show the account id or account.name

Reply:
I'll look at this in the follow up PR